src/UI/SearchPane.re

```diff
@@ -0,0 +1,149 @@
+open Revery;
+open Revery.UI;
+open Revery.UI.Components;
+open Oni_Core;
+open Oni_Model;
+
+module Styles = {
+  let searchPane = (~theme: Theme.t) =>
+    Style.[
+      flexDirection(`Row),
+      height(200),
+      borderTop(~color=theme.sideBarBackground, ~width=1),
+    ];
+
+  let queryPane = (~theme: Theme.t) =>
+    Style.[
+      width(300),
+      borderRight(~color=theme.sideBarBackground, ~width=1),
+    ];
+
+  let resultsPane = Style.[flexGrow(1)];
+
+  let result = (~theme: Theme.t, ~isHovered) =>
+    Style.[
+      flexDirection(`Row),
+      overflow(`Hidden),
+      paddingVertical(4),
+      paddingHorizontal(8),
+      backgroundColor(
+        isHovered ? theme.menuSelectionBackground : Colors.transparentWhite,
+      ),
+    ];
+
+  let fileText = (~font: Types.UiFont.t, ~theme: Theme.t) =>
+    Style.[
+      fontFamily(font.fontFile),
+      fontSize(font.fontSize),
+      color(theme.editorActiveLineNumberForeground),
+    ];
+
+  let lineText = (~font: Types.UiFont.t, ~theme: Theme.t) =>
+    Style.[
+      fontFamily(font.fontFile),
+      fontSize(font.fontSize),
+      color(theme.editorForeground),
+    ];
+
+  let row =
+    Style.[flexDirection(`Row), alignItems(`Center), marginHorizontal(8)];
+
+  let title = (~font: Types.UiFont.t) =>
+    Style.[
+      fontFamily(font.fontFile),
+      fontSize(font.fontSize),
+      color(Colors.white),
+      marginVertical(8),
+      marginHorizontal(8),
+    ];
+
+  let input = (~font: Types.UiFont.t) =>
+    Style.[
+      border(~width=2, ~color=Color.rgba(0., 0., 0., 0.1)),
+      backgroundColor(Color.rgba(0., 0., 0., 0.3)),
+      color(Colors.white),
+      fontFamily(font.fontFile),
+      fontSize(font.fontSize),
+      flexGrow(1),
+    ];
+
+  let clickable = Style.[cursor(Revery.MouseCursors.pointer)];
+};
+
+let%component make =
+              (~theme: Theme.t, ~font: Types.UiFont.t, ~state: Search.t, ()) => {
+  let%hook (hovered, setHovered) = Hooks.state(-1);
+
+  let result = (i, match: Ripgrep.Match.t) => {
+    let onMouseOver = _ => setHovered(_ => i);
+    let onMouseOut = _ => setHovered(hovered => i == hovered ? (-1) : hovered);
+
+    let directory = Rench.Environment.getWorkingDirectory();
+    let re = Str.regexp_string(directory ++ Filename.dir_sep);
+    let getDisplayPath = fullPath => Str.replace_first(re, "", fullPath);
+
+    <Clickable
+      style=Styles.clickable
+      onClick={() =>
+        GlobalContext.current().dispatch(SearchSelectResult(match))
+      }>
+      <View
+        style={Styles.result(~theme, ~isHovered=hovered == i)}
+        onMouseOver
+        onMouseOut>
+        <Text
+          style={Styles.fileText(~font, ~theme)}
+          text={Printf.sprintf(
```

Comment:
One thing I noticed is that, for very long lines, there was a slowdown - we might want to cap the length of the text (or perhaps Revery could do a better job of not rendering text past the window / current clipping region..) 

I was able to reproduce this by searching for `testing` - it seems like some of the yarn lines are very long and hit this case:
![image](https://user-images.githubusercontent.com/13532591/69387893-1d94f200-0c7c-11ea-8b6e-82dd8f520f9c.png)

Reply:
Ah, yeah I noticed this during some earlier testing, before using FlatList, where just 500 results slowed it down to ~1fps, which I thought it should have handled better. Good idea to just cap the length!

Reply:
Is there any way we can calculate a reasonable maximum number of characters based on font size and available width? Perhaps we can estimate an average character width, divide the available width by that and multiply by 1.5 or 2 for a good safety margin?

Reply:
Figured out on Discord I think. We can just use `measuredWidth` from `EditorFont.t`.